tests/integration/agents/supervisor/test_planner.py

```diff
@@ -0,0 +1,191 @@
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content="What is Kubernetes? Explain Kyma function"),
+            ],
+            '{"subtasks": [{"description": "What is Kubernetes?", "assigned_to": "KubernetesAgent"},'
+            '{"description": "Explain Kyma function", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            # tests if a query related to Kyma and Common is divided into the correct subtasks for both agents
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            # tests if a query related to Kyma and Common is divided into the correct subtasks for both agents
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app with python and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app with python", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            # tests if a complex query related to Kyma is divided correctly into two subtasks for the Kyma agent
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="how to enable eventing module and create a subscription for my app?"
+                ),
+            ],
+            '{"subtasks": [{"description": "How to enable eventing module?", "assigned_to": "KymaAgent"},'
+            '{"description": "How to create a subscription for my app?", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+    ],
+)
+def test_invoke_planner(
+    messages,
+    expected_answer,
+    general_query,
+    companion_graph,
+    planner_correctness_metric,
+    answer_relevancy_metric,
+):
+    """Tests the invoke_planner method of SupervisorAgent."""
```

Comment:
Is it from the SupervisorAgent or from the Planner?

Reply:
maybe more specifically:
```yaml
Tests that the invoke_planner method of the Planner either answers simple queries directly or comes up with the expected subtasks.
```

Reply:
I've added more details below.